### PR TITLE
Add game state service

### PR DESCRIPTION
**Pull request by sondrefj**
Replaced old page navigation logic with a new game state service that makes it much easier for developers to add new pages to the game.
Moved game state management from the multiplayerService to gamestate service.

**How it works:**
- add a page name to GAMESTATE enum
- use the goToPage(<GAMESTATE>) method that is in the GameStateService class
- in the parent container (eg. game-component.*), subscribe to changes on the gameStateService instance
- update the template logic to include the new page (eg. switch case)

**Notes**
- This PR does not refactor the logic that is inside the drawing service class

### CODE REVIEW

File: src/app/game/services/game-state-service.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { GAMESTATE } from '@/app/shared/models/interfaces';
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { GameConfigService } from './game-config.service';
+
+enum GAMEMODE {
+  singleplayer = 'SINGLEPLAYER',
+  multiplayer = 'MULTIPLAYER',
+  notSet = 'NOTSET',
+}
```
Comment (andreasrberg) [inline src/app/game/services/game-state-service.ts:10]: Have this here or in interfaces?

**Review by andreasrberg — APPROVED** [review]
Awesome for further development to have this service!

File: src/app/game/game-pick-difficulty/game-pick-difficulty.component.ts [hunk 2]
```diff
@@ -31,31 +29,23 @@ export class GamePickDifficultyComponent implements OnInit {
...
     private translationService: TranslationService
   ) {}
 
   ngOnInit(): void {
     this.gameStateService.savePageToLocalStorage(GAMESTATE.difficultyPicker);
-    if (this.router.url === `/${routes.SINGLEPLAYER}`) {
-      this.isSinglePlayer = true;
+    this.translationService.loadTranslations(this.translationService.getCurrentLang()).subscribe();
+    this.isSinglePlayer = this.gameStateService.isSingleplayer();
+    this.isMultiPlayer = this.gameStateService.isMultiplayer();
```
Comment (andreasrberg) [inline src/app/game/game-pick-difficulty/game-pick-difficulty.component.ts]: Vi er vel enige om å ikke ha difficulty selection i multiplayer?

**Review by andreasrberg — APPROVED** [review]
Looks good! Eventually some bloat with added skeleton for multiplayer in difficulty selection, look at the comment.

File: src/app/game/services/game-state-service.ts (hunk 1 above)
Comment (sondrefj) [reply src/app/game/services/game-state-service.ts:10]: Holdes her frem til den eventuelt blir brukt i noen andre klasser.

File: src/app/game/game-pick-difficulty/game-pick-difficulty.component.ts (hunk 2 above)
Comment (sondrefj) [reply src/app/game/game-pick-difficulty/game-pick-difficulty.component.ts]: Fjernet.

**Review by andreasrberg — APPROVED** [review]

**Pull request merged**